### PR TITLE
Change azure default deployment mode to incremental

### PR DESCRIPTION
**Pull request by gregoryb**
##### ISSUE TYPE

- Bugfix Pull Request
##### COMPONENT NAME

azure-rm-deployment
##### ANSIBLE VERSION

```
ansible 2.1.1.0 and devel as of 372828d
```
##### SUMMARY

COMPLETE deployment mode by default is too dangerous. No one expects to see its cluster being fully destroyed when the default deployment mode in Azure-cli is INCREMENTAL.

### CODE REVIEW

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by devigned** [comment]
Couldn't agree more. Incremental is the way to go lest you destroy all items in your resource group that are not also in your deployment template.

**Comment by resmo** [comment]
The change must not only be made in the docs, but here https://github.com/gregoryb/ansible-modules-extras/blob/957f9770c2497bf92d318ef048491f5f0d972f6a/cloud/azure/azure_rm_deployment.py#L408

**Comment by gregdek** [comment]
Thanks @gregoryb. To the current maintainers, @devigned, @lmazuel please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregoryb** [comment]
Would you like i do the change suggested by @resmo ?

**Comment by devigned** [comment]
ship

**Comment by devigned** [comment]
shipit

File: cloud/azure/azure_rm_deployment.py [hunk 1]
```diff
@@ -42,7 +42,7 @@
       - In incremental mode, resources are deployed without deleting existing resources that are not included in the template. 
         In complete mode resources are deployed and existing resources in the resource group not included in the template are deleted.
```
Comment (resmo) [inline cloud/azure/azure_rm_deployment.py]: Please add some notes here that the default hast changed: e.g. `- The default changed in version 2.3 from C(complete) to C(incremental).`

**Comment by gregdek** [comment]
Thanks again to @gregoryb for this PR, and thanks @devigned, @lmazuel for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/azure/azure_rm_deployment.py (hunk 1 above)
Comment (resmo) [inline cloud/azure/azure_rm_deployment.py]: nvm as it is a fix...

**Pull request merged**